locations/spiders/costacoffee_gb.py

```diff
@@ -1,111 +1,146 @@
 # -*- coding: utf-8 -*-
-import csv
 import scrapy
 
-from locations.items import GeojsonPointItem
+from locations.dict_parser import DictParser
 from locations.hours import OpeningHours
+from locations.geo import postal_regions
 
 
 class CostaCoffeeGBSpider(scrapy.Spider):
     name = "costacoffee_gb"
     item_attributes = {"brand": "Costa Coffee", "brand_wikidata": "Q608845"}
-    allowed_domains = ["www.costa.co.uk"]
-    download_delay = 1
+    download_delay = 1.0
+    # Do our own de-duping as so common!
+    my_ids = []
```

Comment:
Because the fine grain nature of the requests generates a lot of dupes and I do not wish to see them in the console log.

Reply:
Ok. Let's fix the logging then instead of duplicating behavior.

Also, lots of dupes means we're making more requests to the website than we need to. Can we be more efficient with requests?

Reply:
I'm not quite sure how we fix the logging as that is in framework code. All I;m doing is adding a couple of lines of code. I usually remove many many lines i.e. let's not waste words on this.

The costacoffee_gb spider has had a number of iterations. By me, by CJ, by others. CJ has reached out to Costa for a more efficient "in". We are where we are this is as good as it gets for now.

Once again we hope to have not too many of these. On the bright side it is generating 1 HTTP request for ~4 POIs which is a lot better than many .... also JSON is cheaper to server than HTML.